x-pack/test_serverless/functional/test_suites/common/data_usage/intercept_request.ts

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { WebDriver } from 'selenium-webdriver';
+
+export interface ResponseFactory {
+  fail: (reason?: string) => [string, Record<string, any>];
+  fulfill: (responseOptions: Record<string, any>) => [string, Record<string, any>];
+}
+
+export async function interceptRequest(
+  driver: WebDriver,
+  pattern: string,
+  onIntercept: (responseFactory: ResponseFactory) => [string, Record<string, any>],
+  cb: () => Promise<void>
+) {
+  const connection = await driver.createCDPConnection('page');
+
+  return new Promise<void>((resolve, reject) => {
+    connection._wsConnection.on('message', async (data: Buffer) => {
+      const parsed = JSON.parse(data.toString());
+
+      if (parsed.method === 'Fetch.requestPaused') {
+        await new Promise((innerResolve) =>
+          connection.execute(
+            ...onIntercept({
+              fail: () => [
+                'Fetch.failRequest',
+                { requestId: parsed.params.requestId, errorReason: 'Failed' },
+              ],
+              fulfill: (responseOptions: any) => [
+                'Fetch.fulfillRequest',
+                {
+                  requestId: parsed.params.requestId,
+                  ...responseOptions,
+                },
+              ],
+            }),
+            innerResolve
+          )
+        );
+
+        await new Promise((innerResolve) => connection.execute('Fetch.disable', {}, innerResolve));
+        resolve();
+      }
+    });
+
+    new Promise((innerResolve) =>
+      connection.execute(
+        'Fetch.enable',
+        {
+          patterns: [{ urlPattern: pattern }],
+        },
+        innerResolve
+      )
+    )
+      .then(() => {
+        return cb();
+      })
+      .catch((err: Error) => {
+        reject(err);
+      });
+  });
+}
```

Comment:
I think it is a great addition to FTR and should live in [browser](https://github.com/elastic/kibana/blob/6e57a23d18d949e7ac3118c1f27550e2ba9eedf8/packages/kbn-ftr-common-functional-ui-services/services/browser.ts) service. Thoughts?

Reply:
@dmlemeshko I added it to the browser service with some tweaks for error handling, what do you think?  

I believe this was originally written by @dgieselaar and we're using it in the Obs AI Assistant.  I can update it there in a separate PR.